demo-app/src/main/java/io/opentelemetry/android/demo/clients/RecommendationService.kt

```diff
@@ -0,0 +1,27 @@
+package io.opentelemetry.android.demo.clients
+
+import io.opentelemetry.android.demo.model.Product
+import io.opentelemetry.android.demo.ui.shop.cart.CartViewModel
+
+class RecommendationService(
+    private val productCatalogClient: ProductCatalogClient,
+    private val cartViewModel: CartViewModel
+) {
+
+    private fun getAllNonCartProducts(): List<Product>{
+        val allProducts = productCatalogClient.get()
+        val cartItems = cartViewModel.cartItems.value
+
+        return allProducts.filter { product -> cartItems.none { it.product.id == product.id } }
+    }
```

Comment:
Another small nit -- call flow should generally go from top to bottom, so this method should be moved below the two methods where it is called. It's kinda silly, but it helps the reader to follow a vertical flow, rather than having to scroll or move their eyes up and down to follow.